### PR TITLE
Fix react-native-git-upgrade by get package.json instead of selective field

### LINKED ISSUE #11330: react-native-git-upgrade fails on npm v2
The `react-native-git-upgrade` tool is failing to upgrade one of my projects with the following error:

```
Error: The specified version of React Native undefined doesn't exist.
Re-run the react-native-git-upgrade command with an existing version,
for example: "react-native-git-upgrade 0.38.0",
or without arguments to upgrade to the latest: "react-native-git-upgrade".
```

This occurs when running `react-native-git-upgrade` by itself, as well as when a version is specified (e.g. `react-native-git-upgrade 0.39.0`).

I noticed that npm is not returning a JSON response to [`npm view react-native@latest peerDependencies.react version --json`](https://github.com/facebook/react-native/blob/59dd2133cb2bc8292ae932b77337c2a4ba35645b/react-native-git-upgrade/cliEntry.js#L261) as the CLI seems to expect:

```
$ npm view react-native@latest peerDependencies.react version --json
peerDependencies.react = "~15.4.0-rc.4"
version = "0.39.0"
```

This occurred while using node v4.4.7 and npm v2.15.8. Switching to node v6.5.0 and npm v3.10.3 resolved the issue.

## Environment

react-native-git-upgrade: 0.2.3
react-native: 0.34.1
react: 15.3.2
npm: 2.15.8
node: 4.4.7
os: macOS Sierra 10.12.1
react-native latest as of this writing: 0.39.0

## Logs

`react-native-git-upgrade` output:

```
$ react-native-git-upgrade
git-upgrade info Check for updates 
git-upgrade WARN Check for latest version failed Invalid Version: null
Cannot parse yarn version: Using globally installed version of Yarn
0.17.8
git-upgrade info Read package.json files 
git-upgrade info Check declared version 
git-upgrade info Check matching versions 
git-upgrade info Check React peer dependency 
git-upgrade info Check that Git is installed 
git-upgrade info Get information from NPM registry 
git-upgrade ERR! An error occurred during upgrade: 
git-upgrade ERR! Error: The specified version of React Native undefined doesn't exist.
Re-run the react-native-git-upgrade command with an existing version,
for example: "react-native-git-upgrade 0.38.0",
or without arguments to upgrade to the latest: "react-native-git-upgrade".
    at parseInformationJsonOutput (/Users/hramos/.nvm/versions/node/v4.4.7/lib/node_modules/react-native-git-upgrade/cliEntry.js:109:11)
    at Object.<anonymous> (/Users/hramos/.nvm/versions/node/v4.4.7/lib/node_modules/react-native-git-upgrade/cliEntry.js:263:48)
    at next (native)
    at step (/Users/hramos/.nvm/versions/node/v4.4.7/lib/node_modules/react-native-git-upgrade/cliEntry.js:182:191)
    at /Users/hramos/.nvm/versions/node/v4.4.7/lib/node_modules/react-native-git-upgrade/cliEntry.js:182:361
    at tryCallOne (/Users/hramos/.nvm/versions/node/v4.4.7/lib/node_modules/react-native-git-upgrade/node_modules/promise/lib/core.js:37:12)
    at /Users/hramos/.nvm/versions/node/v4.4.7/lib/node_modules/react-native-git-upgrade/node_modules/promise/lib/core.js:123:15
    at flush (/Users/hramos/.nvm/versions/node/v4.4.7/lib/node_modules/react-native-git-upgrade/node_modules/promise/node_modules/asap/raw.js:50:29)
    at nextTickCallbackWith0Args (node.js:420:9)
    at process._tickCallback (node.js:349:13) 
```

### LINKED ISSUE #11334: Upgrade throws version number error
The upgrade didn't work. Throws "The specified version of React Native 0.38.0 doesn't exist". Tried with and without version number.
Currently on 0.37.0.

image

### PR DESCRIPTION
**Pull request by n3tr**
Explain the **motivation** for making this change. What existing problem does the pull request solve?

To fixes #11330 and fixes #11334 

Regarding to this command (using in `react-native-git-upgrade` - [cliEntry.js#L261](https://github.com/facebook/react-native/blob/master/react-native-git-upgrade/cliEntry.js#L261))
```bash
npm view react-native@latest peerDependencies.react version --json
```
The result from this command are not consistent across a different npm versions.

We better get a whole `package.json` file by calling (remove field and subfield option)

```
npm view react-native@latest  --json
``` 

**Test plan (required)**

**Setup**
- Publish `react-native-git-upgrade` to `sinopia`
- `npm install -g react-native-git-upgrade`
- Test against multiple npm versions (`2.15.8` or `3.7.5` or `3.10.10` or ` 4.0.3`)
- `react-native init AwesomeApp  --version 0.38.0 && cd AwesomeApp`
- `react-native-git-upgrade` and `react-native-git-upgrade 0.39.0` should be working properly

---

~~Regarding to this command (using in react-native-git-upgrade - cliEntry.js#L261)~~
~~doesn't produce a json output in some versions of `npm` (we expect it to be a json). I've tested on `3.7.5` and `2.15.8` and the output was~~

```
peerDependencies.react = "~15.4.0-rc.4"
version = "0.39.0"
```

~~This causes `react-native-git-upgrade` failed to parse json and throws an error to the users. (as #11330 and #11334)~~

~~I've also tested on node 7.2.1 + npm 3.10.10 which works fine with current `react-native-git-upgrade` script and the output from `npm view` was~~
```
{
  "peerDependencies.react": "~15.4.0-rc.4",
  "version": "0.39.0"
}
```

### CODE REVIEW

**Comment by ncuillery** [comment]
@n3tr thanks for the PR.

Yes, the `--json` is actually far more recent than I thought :(

IMO, there is no need to fallback on Regexp if the `JSON.parse`fails. It causes unnecessary complexity. We could simply abandon the `--json` option.

There are some cons though:
- Regexp are scary 👻  (I mean, hard to read)
- They are not future-proof, any spaces removal around the equal sign, any single quote instead of double quote and we're dead. 

How about simply removing the requested fields in the command ? NPM would consistently return the whole `package.json` file, no matter the version (I hope). We could continue to use `JSON.parse` safely.

It means calling:

```javascript
npm view react-native@latest
```
instead of

```javascript
npm view react-native@latest version peerDependencies.react
```

**Comment by ncuillery** [comment]
Actually, what I said about single/double quotes is already a reality.vNPM 3.10.9 displays single quote:

```shell
$ npm view react-native@latest version peerDependencies.react         
version = '0.39.0'
peerDependencies.react = '~15.4.0-rc.4'
```

**Comment by n3tr** [comment]
@ncuillery Yes,I Agree - Regex is so scary 😈 

Actually, I have done it that way before I change it to use regex. (still I don't understand why I change it 🤔  )

I'll make it to use a whole `package.json` and try to test it with multiple npm versions.

**Comment by n3tr** [comment]
we can't remove `--json` option.

Without it, the result will be a javascript object instead of JSON string.

**Comment by n3tr** [comment]
@ncuillery 

Done and tested on `2.15.8`, `3.7.5`, `3.10.9`, `3.10.10`, `4.0.3`

File: react-native-git-upgrade/package.json [hunk 1]
```diff
@@ -19,6 +19,7 @@
     "babel-preset-es2015-node": "^6.1.1",
     "babel-preset-stage-3": "^6.17.0",
     "babel-register": "^6.18.0",
+    "minimist": "^1.2.0",
```
Comment (n3tr) [inline react-native-git-upgrade/package.json]: minimist is required in [`index.js:12`](https://github.com/facebook/react-native/blob/master/react-native-git-upgrade/index.js#L12)

**Comment by ncuillery** [comment]
So, `--json` has no effect on the view command with a field filter but works well when requesting the whole view since a long time. Great !

Please, you can bump the version in `package.json` to `0.2.4`too ?

Then, it will LGTM 👍

**Comment by n3tr** [comment]
Updated

**Comment by mkonicek** [comment]
Thanks for fixing this 👍

**Comment by ncuillery** [comment]
@hramos, @scazzy, [0.2.4](https://www.npmjs.com/package/react-native-git-upgrade) has been released.

It should be OK now using NPM2. Thanks @n3tr

**Comment by hramos** [comment]
Thank you!

**Pull request closed without merging**